test/e2e/federatedhpa_test.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2024 The Karmada Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/utils/pointer"
+
+	autoscalingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/autoscaling/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	testhelper "github.com/karmada-io/karmada/test/helper"
+)
+
+var _ = ginkgo.Describe("testing for FederatedHPA and metrics-adapter", func() {
+	var namespace string
+	var deploymentName, serviceName, policyName, federatedHPAName, pressureToolPodName, targetCluster string
+	var deployment *appsv1.Deployment
+	var service *corev1.Service
+	var policy *policyv1alpha1.ClusterPropagationPolicy
+	var federatedHPA *autoscalingv1alpha1.FederatedHPA
+	var pressureTool *corev1.Pod
+
+	ginkgo.BeforeEach(func() {
+		namespace = testNamespace
+		deploymentName = deploymentNamePrefix + rand.String(RandomStrLength)
+		serviceName = deploymentName
+		policyName = deploymentName
+		federatedHPAName = deploymentName
+		pressureToolPodName = deploymentName
+		targetCluster = framework.ClusterNames()[0]
+
+		deployment = testhelper.NewDeployment(namespace, deploymentName)
+		deployment.Spec.Replicas = pointer.Int32(1)
+		deployment.Spec.Template.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+			Limits: map[corev1.ResourceName]resource.Quantity{
+				corev1.ResourceCPU:    resource.MustParse("20m"),
+				corev1.ResourceMemory: resource.MustParse("20Mi"),
+			}}
+		service = testhelper.NewService(namespace, serviceName, corev1.ServiceTypeNodePort)
+		pressureTool = testhelper.NewPod(namespace, pressureToolPodName)
+		pressureTool.Spec.Containers = []corev1.Container{
+			{
+				Name:    pressureToolPodName,
+				Image:   "alpine",
```

Comment:
I'd suggest using a fixed image version here.

Reply:
any suggested version?

Reply:
is `alpine:3.19.1` ok?